src/posthog-core.ts

```diff
@@ -379,6 +380,12 @@ export class PostHog {
             logger.critical('You must name your new library: init(token, config, name)')
             return
         }
+
+        if (_isUndefined(token) || _isEmptyString(token)) {
+            logger.critical('You must provide a token to initialize posthog')
+            return
+        }
```

Comment:
> a user was unintentionally initializing posthog with the empty string as a token, and the empty string as an api_host

should we verify the `api_host` as well? the check is only for `token`.

Reply:
The `api_host` we can also check if it's a valid `URL` as well.

Reply:
I can almost imagine a situation where the empty string is valid (A single page, a single URL, posthog hosted on the same url so `/decide` is always valid) - very edge case but 🙈 

Will ignore URL validation here to keep the PR smaller :)